src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -461,7 +464,20 @@ VtolAttitudeControl::Run()
 		_local_pos_sub.update(&_local_pos);
 		_local_pos_sp_sub.update(&_local_pos_sp);
 		_pos_sp_triplet_sub.update(&_pos_sp_triplet);
-		_airspeed_validated_sub.update(&_airspeed_validated);
+		bool got_airspeed = _airspeed_validated_sub.update(&_airspeed_validated);
+		if (PX4_ISFINITE(_calibrated_airspeed_m_s_smoothed) && _params.vt_aspd_tau > 0.1f) {
+			// Smooth the airspeed value using exponential moving average
+			float alpha = 1 - expf(-dt_us / (_params.vt_aspd_tau * 1e6f));
+			_calibrated_airspeed_m_s_smoothed = _calibrated_airspeed_m_s_smoothed * (1-alpha) + _airspeed_validated.true_airspeed_m_s * alpha;
+		} else if (got_airspeed) {
+			// Use the sensor value
+			_calibrated_airspeed_m_s_smoothed = _airspeed_validated.true_airspeed_m_s;
```

Comment:
```suggestion
			_calibrated_airspeed_m_s_smoothed = _airspeed_validated.calibrated_airspeed_m_s;
```

Reply:
A side effect of this if-else is that whenever the parameter value is below 0.1 and airspeed_validated gives publications, then _calibrated_airspeed_m_s_smoothed will perfectly track airspeed_validated, and be re-published as airspeed_smoothed, but there will be no smoothing. Is this intentional?

Reply:
See above